### PR TITLE
[FLINK-22320][docs][table] Add documentation for new introduced ALTER TABLE statements

### PR DESCRIPTION
**Pull request by LadyForest**
## What is the purpose of the change

This PR is a subtask of FLINK-21634 to add documentation.

## Brief change log

Add documentation for `ALTER TALBE [IF EXISTS] <table_identifier>` statements.

## Verifying this change

This change can be verified by building the document locally at http://localhost:1313

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented?  docs

### CODE REVIEW

**Comment by LadyForest** [comment]
@flinkbot run azure

File: docs/content/docs/dev/table/sql/alter.md [hunk 1]
```diff
@@ -125,38 +179,310 @@ tables = table_env.list_tables()
...
+- Add a computed column which is derived from another computed or nonexistent columns
+- Add a computed column with invalid expression
+
+**Add Table Constraint**
+
+```text
+ALTER TABLE [IF EXISTS] [catalog_name.][db_name.]table_name ADD [CONSTRAINT constraint_name] PRIMARY KEY(col1[, col2, ...]) NOT ENFORCED
+```
+
+Add primary key to the table. Throws `ValidationException` when either of following conditions are met
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:360]: It's better if we can refer to the create#PRIMARY KEY section. I think we have the same rules when creating a table with pk.

File: docs/content/docs/dev/table/sql/alter.md [hunk 2]
```diff
@@ -125,38 +179,310 @@ tables = table_env.list_tables()
...
+**Add Table Watermark**
+
+```text
+ALTER TABLE [IF EXISTS] [catalog_name.][db_name.]table_name ADD WATERMARK FOR rowtime_column_name AS watermark_strategy_expression
+```
+
+Add a watermark to the table. Add primary key to the table. Throws `ValidationException` when either of following conditions are met
+- The table already defines watermark 
+- The specified row-time field does not exist
+- The watermark strategy expression is invalid.
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:375]: Do we need to write these details in the doc... I think we can remove this.

**Review by fsk119 — COMMENTED** [review]
Thank for your contribution. I left some comments.

File: docs/content/docs/dev/table/sql/alter.md [hunk 3]
```diff
@@ -125,38 +179,258 @@ tables = table_env.list_tables()
...
+    ts AS TO_TIMESTAMP(log_ts) AFTER log_ts,
+    PRIMARY KEY (id) NOT ENFORCED,
+    WATERMARK FOR ts AS ts - INTERVAL '3' SECOND
+);
+```
+
+### MODIFY
+Use `MODIFY` clause to change column's position, type, comment or nullability, change primary key columns and watermark strategy to an existing table. 
+
+To modify an existent column to a new position, use `FIRST` or `AFTER col_name`.
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:362]: Describe the default behaviour.

File: docs/content/docs/dev/table/sql/alter.md [hunk 4]
```diff
@@ -125,38 +179,258 @@ tables = table_env.list_tables()
...
+ALTER TABLE MyTable ADD category_id STRING COMMENT 'identifier of the category';
+
+-- add columns, constraint, and watermark
+ALTER TABLE MyTable ADD (
+    log_ts STRING COMMENT 'log timestamp string' FIRST,
+    ts AS TO_TIMESTAMP(log_ts) AFTER log_ts,
+    PRIMARY KEY (id) NOT ENFORCED,
+    WATERMARK FOR ts AS ts - INTERVAL '3' SECOND
+);
+```
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:357]: I think the origin notes is very nice. Can we add it back? 

```
<span class="label label-danger">Note</span> Add a column to be primary key will change the column's nullability to false implicitly.

```

File: docs/content/docs/dev/table/sql/alter.md [hunk 5]
```diff
@@ -125,38 +179,258 @@ tables = table_env.list_tables()
...
+-- modify a column type, comment and position
+ALTER TABLE MyTable MODIFY measurement double COMMENT 'unit is bytes per second' AFTER `id`;
+
+-- modify definition of column log_ts and ts, primary key, watermark. They must exist in table schema
+ALTER TABLE MyTable MODIFY (
+    log_ts STRING COMMENT 'log timestamp string' AFTER `id`,  -- reorder columns
+    ts AS TO_TIMESTAMP(log_ts) AFTER log_ts,
+    PRIMARY KEY (id) NOT ENFORCED,
+    WATERMARK FOR ts AS ts -- modify watermark strategy
+);
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:376]: add the origin notes:

```
<span class="label label-danger">Note</span> Modify a column to be primary key will change the column's nullability to false implicitly.
```

File: docs/content/docs/dev/table/sql/alter.md [hunk 6]
```diff
@@ -125,38 +179,258 @@ tables = table_env.list_tables()
...
+ALTER TABLE MyTable DROP WATERMARK;
 ```
 
+### RENAME
+Use `RENAME` clause to rename column or an existing table.
+
+The following examples illustrate the usage of the `RENAME` statements.
+```sql
+-- rename column
+ALTER TABLE MyTable RENAME `data` TO payload;
```
Comment (fsk119) [inline docs/content/docs/dev/table/sql/alter.md:404]: nit: 
```
`data` -> data
```

**Review by fsk119 — COMMENTED** [review]
Thanks for your update. I left some minor suggestions.

File: docs/content/docs/dev/table/sql/alter.md (hunk 6 above)
Comment (LadyForest) [reply docs/content/docs/dev/table/sql/alter.md:404]: > nit:
> 
> ```
> `data` -> data
> ```

`DATA` is a reserved keyword, https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/table/sql/overview/

**Review by fsk119 — APPROVED** [review]
Thanks for your update. LGTM

**Pull request closed without merging**